core/src/options/from_meta.rs

```diff
@@ -33,6 +35,26 @@ impl ParseData for FromMetaOptions {
     fn parse_field(&mut self, field: &syn::Field) -> Result<()> {
         self.base.parse_field(field)
     }
+
+    fn validate_body(&self, errors: &mut Accumulator) {
+        match self.base.data {
+            Data::Enum(ref data) => {
+                // Adds errors for duplicate `#[darling(word)]` annotations across all variants.
+                let word_variants: Vec<_> = data
+                    .iter()
+                    .filter(|variant| variant.word.is_some())
+                    .collect();
+                if word_variants.len() > 1 {
+                    for variant in word_variants[1..].iter() {
+                        if let Some(word) = variant.word {
+                            errors.push(Error::duplicate_field("word").with_span(&word.span()));
```

Comment:
Rather than `duplicate_field` here, let's use a custom error:

```suggestion
                            errors.push(Error::custom("`word` can only be applied to one variant").with_span(&word.span()));
```